core/chains/terra/terratxm/key_wrapper.go

```diff
@@ -0,0 +1,56 @@
+package terratxm
+
+import (
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/terrakey"
+	"github.com/smartcontractkit/terra.go/key"
+)
+
+var _ key.PrivKey = KeyWrapper{}
+
+// KeyWrapper wrapper around a terra transmitter key
+// for use in the terra txbuilder and client.
+type KeyWrapper struct {
+	key terrakey.Key
+}
+
+// NewKeyWrapper create a key wrapper
+func NewKeyWrapper(key terrakey.Key) KeyWrapper {
+	return KeyWrapper{key: key}
+}
+
+// Reset nop
+func (k KeyWrapper) Reset() {}
+
+// ProtoMessage nop
+func (k KeyWrapper) ProtoMessage() {}
+
+// String nop
+func (k KeyWrapper) String() string {
+	return ""
+}
```

Comment:
Is this all prod-ready code? Seems unfinished.

Reply:
Yeah its ready, unfortunately the txbuilder library that we use for signing needs the key to implement the PrivKey interface, but only uses the Sign/Pubkey/Equals methods. It stems from type reuse in the library. Ill add a comment